src/sentry/integrations/source_code_management/commit_context.py

```diff
@@ -409,3 +422,224 @@ def update_comment(
+    @abstractmethod
+    def organization_option_key(self) -> str:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def referrer(self) -> Referrer:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def referrer_id(self) -> str:
+        raise NotImplementedError
+
+    @abstractmethod
+    def queue_task(self, pr: PullRequest, project_id: int) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_comment_body(self, issue_ids: list[int]) -> str:
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_comment_data(
+        self,
+        organization: Organization,
+        repo: Repository,
+        pr: PullRequest,
+        comment_body: str,
+        issue_ids: list[int],
+    ) -> dict[str, Any]:
+        raise NotImplementedError
+
+    def get_issue_ids_from_pr(self, pr: PullRequest, limit: int = MAX_SUSPECT_COMMITS) -> list[int]:
+        with connection.cursor() as cursor:
+            cursor.execute(
+                """
+                SELECT go.group_id issue_id
+                FROM sentry_groupowner go
+                JOIN sentry_pullrequest_commit c ON c.commit_id = (go.context::jsonb->>'commitId')::bigint
+                JOIN sentry_pull_request pr ON c.pull_request_id = pr.id
+                WHERE go.type=0
+                AND pr.id=%s
+                ORDER BY go.date_added
+                LIMIT %s
+                """,
+                params=[pr.id, limit],
+            )
+            return [issue_id for (issue_id,) in cursor.fetchall()]
+
+    def get_top_5_issues_by_count(
+        self, issue_ids: list[int], project: Project
+    ) -> list[dict[str, Any]]:
+        """Given a list of issue group ids, return a sublist of the top 5 ordered by event count"""
+        request = SnubaRequest(
+            dataset=Dataset.Events.value,
+            app_id="default",
+            tenant_ids={"organization_id": project.organization_id},
+            query=(
+                Query(Entity("events"))
+                .set_select([Column("group_id"), Function("count", [], "event_count")])
+                .set_groupby([Column("group_id")])
+                .set_where(
+                    [
+                        Condition(Column("project_id"), Op.EQ, project.id),
+                        Condition(Column("group_id"), Op.IN, issue_ids),
+                        Condition(Column("timestamp"), Op.GTE, datetime.now() - timedelta(days=30)),
+                        Condition(Column("timestamp"), Op.LT, datetime.now()),
+                        Condition(Column("level"), Op.NEQ, "info"),
+                    ]
+                )
+                .set_orderby([OrderBy(Column("event_count"), Direction.DESC)])
+                .set_limit(5)
+            ),
+        )
+        return raw_snql_query(request, referrer=self.referrer.value)["data"]
+
+
+def run_pr_comment_workflow(integration_name: str, pr_id: int, project_id: int) -> None:
```

Comment:
can we move the `run_pr_comment_workflow` function logic into the task itself? i don't think we need it to live as a separate function if it only is run in the task

Reply:
@cathteng did you mean `src/sentry/integrations/github/tasks/pr_comment.py` or somewhere else?

Reply:
yes please, since that task is only calling this function

Reply:
@cathteng Ah, my plan is to make it generic so that it can be used by other integrations as well (i.e., GitLab, Bitbucket, Azure).

Reply:
it should be generic because we are using functions from the installation which would be implemented by other providers, right?